immut/sorted_map/sorted_map.mbti

```diff
@@ -23,6 +23,7 @@ impl T {
   from_iter[K : Compare, V](Iter[(K, V)]) -> Self[K, V]
   from_json[V : @json.FromJson](Json) -> Self[String, V]!@json.JsonDecodeError
   insert[K : Compare, V](Self[K, V], K, V) -> Self[K, V] //deprecated
+  intersection[K : Compare, V1, V2](Self[K, V1], Self[K, V2]) -> Iter[(K, V1, V2)]
```

Comment:
The `intersection` should return another sorted_map, similar to the `intersection` methods of `set`, `sorted_set`, and `immut/sorted_set`

Reply:
The result `@immut/sorted_map.T[K, (V1, V2)]` is often useless, as users often want to do some merge work of `V1` and `V2`.

Reply:
`intersection(Map, Map) -> Map` can be implemented in O(m log(n/m + 1)) time complexity, which is better than the current implementation O(m log n). Of course, returning a map will cause more space overhead, but users who only iterate over the result can always easily optimize it by `map1.iter().filter(existInMap2).each(f)`. 

If it returns an iter, the order information will be lost and it will take O(m log n) + O(k log k) (k < m < n) time to construct an intersection map. Furthermore, you have to break existing set operations to keep API consistent.

Reply:
Agreed. @Yoorkin Could you fire a new PR for `O(m log(n/m + 1))` implementation that returns `Map`?